api/src/main/java/org/openmrs/module/attachments/AttachmentsService.java

```diff
@@ -60,5 +61,13 @@
 	 */
 	List<Attachment> getAttachments(Patient patient, Visit visit, boolean includeVoided);
 	
+	/**
+	 * Get a patient's attachments that are associated with a specified concept.
+	 * 
+	 * @param concept
+	 * @throws APIException if non-complex obs are mistakenly returned
+	 */
+	List<Attachment> getAttachments(Patient patient, Concept concept);
```

Comment:
This is my main issue, the changes to `AttachmentsService` are possibly much wider than thought here.

I believe this is where the existing API could be deprecated and a new one is to be introduced that caters for the new `concept` argument.

Reply:
> I believe this is where the existing API could be deprecated and a new one is to be introduced that caters for the new concept argument.

@mks-d Looks like some good time is required for this one; do you think its okay to handle this in an independent ticket?